modules/apps/frontend-data-set/frontend-data-set-taglib/src/main/resources/META-INF/frontend-data-set.tld

```diff
@@ -185,6 +185,12 @@
 			<required>false</required>
 			<rtexprvalue>true</rtexprvalue>
 		</attribute>
+		<attribute>
+			<name>dynamicFDSFilters</name>
```

Comment:
Can we call this attribute `filters`? I don't think `fds` prefix is really needed, and I don't see anything particularly dynamic about them :slightly_smiling_face: 

Also, I think we can add the same API to the `<classic-display>`. You can place the tag common tag logic in the `BaseDisplayTag`.

Reply:
Hi @markocikos, I chose dynamic as a prefix to differentiate these filters from those provided at these points `_fdsFilterRegistry.getFDSFilters(fdsName)` and also because these filters can be manipulated by the customer.

about `fds` I think it make sense keep it because this is a list of FDSFilter, what do you think?

Reply:
I realized that the `ClassicDisplayTag` doesn't support _fdsFiltersContext yet

Reply:
> I chose dynamic as a prefix to differentiate these filters from those provided at this point `_fdsFilterRegistry.getFDSFilters(fdsName)` and also because these filters can be manipulated by the customer.

@MarinhoFeliphe  I am not familiar with the larger context of your usage of this API. If I'm understanding this correctly, 'manipulated by the customer' comes from this larger context. But, looking at the API in isolation, from FDS perspective, I don't see any increased flexibility, or ability to manipulate objects. In both cases, it is a set of Java classes. It is just passed into the tag through different means, either through registry or directly through an attribute.

To give an example of a truly more flexible API, in DXP we have several tags that support dynamic attributes. This is set with `<dynamic-attributes>true</dynamic-attributes>` in the tld file. With this, you can pass in any attribute and it will   transfer typically to the container of the component.

I fear that setting a `dynamic` prefix to attribute may be misleading, for example a future dev may be thinking you can pass in different types of filters.

On a related note, there is no need to differentiate these filters with filters passed in through registry. They are the same objects, and there is also no overlap in API. Different naming would make sense if there was an overlap, for example, if these were two attributes of the same tag.  

> about `fds` I think it make sense keep it because this is a list of FDSFilter, what do you think?

I am not entirely sure on this one. Generally, API should be abstracted. The name of the attribute does not need to reflect data type. For example, we call it `id`, not `stringId`. We also don't need to prefix attributes with something that distinguishes them for that tag. For example, we call it `id`, not `fdsId`. The only reason to place a prefix is if we think omitting it may cause confusion. For example, if devs commonly use some `Filter` object in the same module as `FDSFilter`. Is this the case? Technically [yes](https://github.com/liferay/liferay-portal/blob/master/util-java/src/com/liferay/util/service/ServiceProps.java), but this is not a common case. I don't see the prefix needed here, but it's a close and subjective call. I would be ok will keeping it.

Reply:
> I realized that the ClassicDisplayTag doesn't support `_fdsFiltersContext` yet

It should! This is something we will add, sooner or later. If adding this support goes too much beyond the scope of this task, we can leave it for a separate task.